### PR TITLE
2 autogenerate a unique qr code

### PR DESCRIPTION
**Pull request by yifeng15**
add_edit_event.xml is wrong. Need to fix it and implement the adding event.

### CODE REVIEW

File: app/src/main/java/com/example/scanpal/AddEditEventFragment.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.example.scanpal;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.fragment.app.Fragment;
+
+public class AddEditEventFragment extends Fragment {
```
Comment (Logan9312) [inline app/src/main/java/com/example/scanpal/AddEditEventFragment.java:10]: Needs docstring for class

File: app/src/main/java/com/example/scanpal/QRCodeGenerator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import com.google.zxing.EncodeHintType;
+import com.google.zxing.WriterException;
+import com.google.zxing.common.BitMatrix;
+import com.google.zxing.qrcode.QRCodeWriter;
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class QRCodeGenerator {
```
Comment (Logan9312) [inline app/src/main/java/com/example/scanpal/QRCodeGenerator.java:16]: Needs docstring for class

File: app/src/main/java/com/example/scanpal/EventController.java [hunk 3]
```diff
@@ -14,14 +25,17 @@
...
     private final FirebaseFirestore database; // instance of the database
+    private final FirebaseStorage storage;
 
     /**
      * Constructs an EventController with a reference to a Firestore database.
      *
      * @param database The Firestore database instance used for event operations.
      */
     public EventController(FirebaseFirestore database) {
+        // Why not just call FirebaseFireStore.getInstance()?
```
Comment (Logan9312) [inline app/src/main/java/com/example/scanpal/EventController.java:36]: Agreed, we should do this

**Review by Logan9312 — CHANGES_REQUESTED** [review]
Everything looks good, just added a couple comments where it seems like docstring is missing for Part 3 deliverables. I couldn't test because the fragment isn't hooked up yet so we can just re-test once more event functionality is complete. Seems good to merge for now.

Not sure if this will conflict with anything @adviislam10 has done though so it's worth discussing with him.

**Comment by yifeng15** [comment]
Had to force push to revert back to the state before mergin with the main.

**Review by Logan9312 — APPROVED** [review]
Looks good to me

**Review by Sundance636 — APPROVED** [review]
Looks good!

**Review by Nandan-18 — APPROVED** [review]
looks good, tested and works as expected!

**Pull request merged**